hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -923,12 +937,123 @@ public <N extends Number> SqmExpression<N> abs(Expression<N> x) {
 		);
 	}
 
+	@Override
+	public JpaExpression<Duration> duration(long magnitude, TemporalUnit unit) {
+		return new SqmToDuration<>(
+				literal( magnitude ),
```

Comment:
I just quickly played with PostgreSQL to see how it behaves, but couldn't see any apparent problems.

```sql
prepare test(interval) as select 
current_timestamp + $1;

execute test(interval '1 days');
```

I would assume other DBs have no trouble with parameters either because they usually have dedicated functions for timestamp arithmetic, which should be well typed by construction.

Worst case, we could pass `SqlAstNodeRenderingMode.NO_PLAIN_PARAMETER` to the `PatternRenderer` for `TimestampaddFunction` to force rendering the cast around the parameters.

Reply:
I mean you can try it for yourself, change `literal()` to `value()` and run the test to see which databases choke.